Disable I2C interface in ICM42688P sensor when it is initialized

jlaitine opened the pull request:
This prevents accidental misconfiguration via I2C if there are multiple SPI devices on the same bus. The device may hear it's i2c address and write some register, while there is data transfer ongoing with another device.

dagar commented on src/drivers/imu/invensense/icm42688p/InvenSense_ICM42688P_registers.hpp:
```diff
@@ -159,6 +159,16 @@ enum SIGNAL_PATH_RESET_BIT : uint8_t {
+	FIFO_COUNT_REC         = Bit6,
+	FIFO_COUNT_ENDIAN      = Bit5,
+	SENSOR_DATA_ENDIAN     = Bit4,
+	UI_SIFS_CFG            = Bit1 | Bit0,
```
How about something like this? We need a better register abstraction...

```suggestion
	UI_SIFS_CFG_DISABLE_I2C            = Bit1 | Bit0,
```

jlaitine replied:
Sure!

jlaitine replied:
I also added the definitions to the copy of the file under boards/modalai/voxl2-slpi/src/drivers/icm42688p just in case. I am not sure how it is used, but I guess it doesn't harm at least, as it didn't exist before.

dagar commented on src/drivers/imu/invensense/icm42688p/ICM42688P.hpp:
```diff
@@ -179,9 +179,10 @@ class ICM42688P : public device::SPI, public I2CSPIDriver<ICM42688P>
+	static constexpr uint8_t size_register_bank0_cfg{17};
 	register_bank0_config_t _register_bank0_cfg[size_register_bank0_cfg] {
 		// Register                              | Set bits, Clear bits
+		{ Register::BANK_0::INTF_CONFIG0,         INTF_CONFIG0_BIT::UI_SIFS_CFG, 0}, // 0x3 disable I2C
```
Can you put this just before INTF_CONFIG1 (0x4D) to keep it in order?

jlaitine replied:
Yes, I intended it to be as early as possible, since it protects the bus. But it doesn't make much difference, it looks better like you suggested. Done

jlaitine commented on src/drivers/imu/invensense/icm42688p/ICM42688P.hpp:
```diff
@@ -179,9 +179,10 @@ class ICM42688P : public device::SPI, public I2CSPIDriver<ICM42688P>
+	static constexpr uint8_t size_register_bank0_cfg{17};
 	register_bank0_config_t _register_bank0_cfg[size_register_bank0_cfg] {
 		// Register                              | Set bits, Clear bits
+		{ Register::BANK_0::INTF_CONFIG0,         INTF_CONFIG0_BIT::UI_SIFS_CFG, 0}, // 0x3 disable I2C
```
My original code just added the bits 0,1 , which disables the I2C, but didn't alter the default bits. I added the other two  bits in here for clarity as you suggested.

jlaitine commented:
@dagar, just checking if this was forgotten, or perhaps not needed in upstream px4? I believe I addressed the comments, so let's merge, modify further or close this PR :slightly_smiling_face:

dagar commented:
Looks good to me, but why is the voxl2-slpi version only partially updated? Hopefully that driver will be eliminated long term, but I think something must be missing in QuRT SPI support.

jlaitine commented:
> Looks good to me, but why is the voxl2-slpi version only partially updated? Hopefully that driver will be eliminated long term, but I think something must be missing in QuRT SPI support.

I didn't update the voxl2 driver as I wasn't sure whether the change was wanted there or not;. Also, I didn't have that hw at hand. It is of course simple to add if wanted.
